.github/workflows/ci.yml

```diff
@@ -0,0 +1,24 @@
+name: CI
+
+on:
+  push:
+    branches: [ 'master' ]
+  pull_request:
+
+env:
+  GRADLE_OPTS: >
+    -Dorg.gradle.daemon=false
+    -Dorg.gradle.vfs.watch=false
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          java-version: 11
+          distribution: 'zulu'
+      - uses: gradle/gradle-build-action@v2
+        with:
+          arguments: build
```

Comment:
Don't think we should use `build` all in one task - since we're running on the hosted runners, pretty sure this task will OOM.

Reply:
Is this a concern for pure Kotlin projects of this size though? I can totally see this OOMs occurring on an Android project.

Reply:
This project does include an Android sample app. Our initial builds started OOMing very quickly.